login_server/loginserver.py

```diff
@@ -326,13 +326,15 @@ def handle_client_message(self, msg):
 
     def handle_http_request_message(self, msg):
         if msg.env['PATH_INFO'] == '/status':
-            self.logger.info('Served status request via HTTP to peer ' + str(msg.peer))
+            if "HTTP_ORIGIN" in msg.env:
```

Comment:
I read that HTTP_ORIGIN is set by the client, which can't be trusted. I also see that REMOTE_ADDR is supposedly the connecting client's IP. Can you check if REMOTE_ADDR works?

Reply:
Good catch, as far as I can tell REMOTE_ADDR does show the connecting client's IP.
I initially dismissed REMOTE_ADDR because it always returned "127.0.0.1" while I was testing it on my local machine. I just tested the connection from another machine on my LAN and REMOTE_ADDR correctly returned the IP of that machine.
See commit 649b1bc below.

Reply:
Do not skip logging if REMOTE_ADDR is not in the environment. Just log "unknown" as the IP or something. Just to make sure we'll always know requests are coming even if we're not able to log from who with the current code.

Reply:
Ok, I see it's done.